composer/trainer/trainer.py

```diff
@@ -944,6 +959,8 @@ def __init__(
                                               strict_model_weights=load_strict_model_weights,
                                               chunk_size=load_chunk_size,
                                               progress_bar=load_progress_bar)
+            # Always override run_name. In the future, we'll use the loaded name and not require run_name
+            self.state.run_name = generated_run_name
```

Comment:
If `autoresume and run_name != self.state.run_name`, can you raise an exception (for the time being), as the run name must be the same for autoresume to work properly?